pslogin/src/main/scala/WorldSessionActor.scala

```diff
@@ -149,6 +150,18 @@ class WorldSessionActor extends Actor with MDCContextAware {
 
   var timeDL : Long = 0
   var timeSurge : Long = 0
+  lazy val unsignedIntMaxValue : Long = Int.MaxValue.toLong * 2L + 1L
+  val sysTime : ()=>Long = {
+    //if the first nanoTime call is negative, subsequent calls will be only "slightly less negative"
+    //even with math.abs, each call to nanoTime would result in time counting down
+    if(System.nanoTime > 0) {
+      System.nanoTime
+    }
```

Comment:
If nanotime is going to be negative, is there any reason we cant just use millis or does this require more testing?

Reply:
In regards to testing, we're not certain what the field calculates to begin with.  I'm just trying to accommodate the original logic under the assumption that, even if it failed, we never saw reprecussions of that failure.  At the same time, I am trying to bring it closer to behavior observed in examined packet captures.

If nanoTime produces one negative number, it will probably keep producing negative numbers for the extent of this run time.

Reply:
Ah I see. When it is negative, the numbers count down. Can you return `- System.nanotime` for the negative case to make it so we're always using nanotime?

Reply:
After thinking about an appropriate rendition for the "if negative values" part, I came to the conclusion that `currentTimeMillis` should be more than apprioriate.